src/test/suite/scroll.test.ts

```diff
@@ -0,0 +1,73 @@
+import * as assert from "assert";
+import * as vscode from "vscode";
+import * as sinon from "sinon";
+import { getCursorlessApi } from "../../util/getExtensionApi";
+import { openNewEditor } from "../openNewEditor";
+
+suite("scroll", async function () {
+  this.timeout("100s");
+  this.retries(3);
+
+  teardown(() => {
+    sinon.restore();
+  });
+
+  test("top whale", topWhale);
+  test("bottom whale", bottomWhale);
+});
+
+async function topWhale() {
+  const graph = (await getCursorlessApi()).graph!;
+  const editor = await openNewEditor("hello\nworld");
+  await graph.hatTokenMap.addDecorations();
+
+  await vscode.commands.executeCommand(
+    "cursorless.command",
+    "crown whale",
+    "scrollToTop",
+    [
+      {
+        type: "primitive",
+        mark: {
+          type: "decoratedSymbol",
+          symbolColor: "default",
+          character: "w",
+        },
```

Comment:
as above, maybe we should just use cursor mark to avoid hat allocation fragility?